docs/backends/common/index.rst

```diff
@@ -0,0 +1,51 @@
+.. _sec-backends-common:
+
+Common Backend Impl (CBI)
+=========================
+
+The common backend implementations are implementations for re-use by other
+backends. For example, instead of the Linux, FreeBSD, and macOS backends all
+implement the same POSIX asynchronous I/O interface, then it is instead provided
+here, and the can all include it in their support for async. interfaces.
+
+Similarly, then **xNVMe** provide several software-only implementations of the
+**xNVMe** API, such as the **thrpool** which implements the asynchronous parts
+of the **xNVMe** API by using a thread-pool.
+
+This composability grants convenience during development and interesting
+experiments when mixing abstractions and system interfaces. In addition to
+providin common implementations, interface emulation, then **xNVMe** also
+provide simple device emulation via the **ramdisk** backend implementation.
+
+Instrumentation
+---------------
+
+* Memory Allocators
+
+  - ``posix```, Use **libc** ``malloc()/free()`` with ``sysconf()`` for
+    alignment, add link to CBI
```

Comment:
What does "add link to CBI" refer to?

Reply:
Ah, I see this is like a TODO. Then it should be removed here.

Reply:
Please skip the backends and tools sections, as they are incomplete and may not make sense. These sections were not addressed due to the substantial size of this PR.